app/js/services/services.js

```diff
@@ -534,6 +534,37 @@ KMCServices.factory('PlayerService', ['$http', '$modal', '$log', '$q', 'apiServi
 				);
 				return deferred.promise;
 			},
+			'playerUpgradeV7': function(playerObj, mode, templateId) {
+				var deferred = $q.defer();
+				var paramsPromise;
+				switch (mode) {
+					case 'default':
+						paramsPromise = PlayerDataService.getDefaultV7PlayerConf(playerObj.name, playerObj.description);
+						break;
+					case 'template':
+						paramsPromise = PlayerDataService.getV7PlayerConfFromTemplate(playerObj.name, templateId);
```

Comment:
from what I understand, you will call `getV7PlayerConfFromTemplate` to get a promise that already has been fired (?).
and then, from line **551** you will have an additional **.then()**?
so, the response to the internal promise, is another promise object?

Reply:
@vladimir-poleh

Reply:
There were some other problems with promises. Now it should work correctly.